lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -429,6 +442,43 @@
       username: 'admin'
       password: 'password'
       verify_ssl: false
+
+- name: Create a new OpenStack Director provider in ManageIQ with rsa keypair
+  manageiq_provider:
+    name: 'EngDirector'
+    type: 'Director'
+    api_version: 'v3'
+    state: 'present'
+    provider:
+      hostname: 'director.example.com'
+      userid: 'admin'
+      password: 'password'
+      security_protocol: 'ssl-with-validation'
+      verify_ssl: 'true'
+      certificate_authority: |
+        -----BEGIN CERTIFICATE-----
+        FAKECERTsdKgAwIBAgIBATANBgkqhkiG9w0BAQsFADAmMSQwIgYDVQQDDBtvcGVu
+        c2hpZnQtc2lnbmVyQDE1MDMzMjAxMTkwHhcNMTcwODIxMTI1NTE5WhcNMjIwODIw
+        MTI1NTIwWjAmMSQwIgYDVQQDDBtvcGVuc2hpZnQtc2lnbmVyQDE1MDMzMjAxMTkw
+        ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDUDnL2tQ2xf/zO7F7hmZ4S
+        ZuwKENdI4IYuWSxye4i3hPhKg6eKPzGzmDNWkIMDOrDAj1EgVSNPtPwsOL8OWvJm
+        AaTjr070D7ZGWWnrrDrWEClBx9Rx/6JAM38RT8Pu7c1hXBm0J81KufSLLYiZ/gOw
+        Znks5v5RUSGcAXvLkBJeATbsbh6fKX0RgQ3fFTvqQaE/r8LxcTN1uehPX1g5AaRa
+        z/SNDHaFtQlE3XcqAAukyMn4N5kdNcuwF3GlQ+tJnJv8SstPkfQcZbTMUQ7I2KpJ
+        ajXnMxmBhV5fCN4rb0QUNCrk2/B+EUMBY4MnxIakqNxnN1kvgI7FBbFgrHUe6QvJ
+        AgMBAAGjIzAhMA4GA1UdDwEB/wQEAwICpDAPBgNVHRMBAf8EBTADAQH/MA0GCSqG
+        SIb3DQEBCwUAA4IBAQAYRV57LUsqznSLZHA77o9+0fQetIE115DYP7wea42PODJI
+        QJ+JETEfoCr0+YOMAbVmznP9GH5cMTKEWHExcIpbMBU7nMZp6A3htcJgF2fgPzOA
+        aTUtzkuVCSrV//mbbYVxoFOc6sR3Br0wBs5+5iz3dBSt7xmgpMzZvqsQl655i051
+        gGSTIY3z5EJmBZBjwuTjal9mMoPGA4eoTPqlITJDHQ2bdCV2oDbc7zqupGrUfZFA
+        qzgieEyGzdCSRwjr1/PibA3bpwHyhD9CGD0PRVVTLhw6h6L5kuN1jA20OfzWxf/o
+        XUsdmRaWiF+l4s6Dcd56SuRp5SGNa2+vP9Of/FX5
+        -----END CERTIFICATE-----
+    auth_type:
+      hostname: 'director.example.com'
+      role: 'ssh_keypair'
```

Comment:
what are hostname & role here?  you didn't document those, only userid, auth_key.

Reply:
I've removed hostname as I cant see any instance where the Ironic node wouldn't be the director node. The role parameter is required to indicate the authentication type. Hopefully an improvement.

Reply:
Awesome, all looks great now, but I think there should be a way to omit role here.
I think if you give Director `ssh_keypair_role="ssh_keypair"` in `supported_providers()`, you will be able to omit it, via this logic:
https://github.com/snecklifter/ansible/blob/2fb8d3b7e139/lib/ansible/modules/remote_management/manageiq/manageiq_provider.py#L637

Reply:
Thanks, good point regarding the supported_providers logic - have updated.

I'm not sure why my testing showed it was ok to drop hostname but it has required=True so have re-added it. It should always be the same as the provider hostname though so perhaps there is a better way to do this?

Reply:
@snecklifter does this provider even need an ssh_keypair Endpoint?  I suspect it only needs an Authentication.
@yaacov, is there a way to make hostname unnecessary here?  Is there a way to not create hostname?

Reply:
shipit
This is great, can improve in followup PR if we find how.